### PR TITLE
chore: replace structopt with clap3

### PR DESCRIPTION
**Pull request by clifton**
structopt is deprecated in favor of clap 3

todo:
- [x] https://github.com/roynalnaruto/svm-rs/pull/14
- [x] update docs
- [x] add shell completions to cast

### CODE REVIEW

**Comment by clifton** [comment]
can also remove some of the clap features (tho the additional dependencies are minimal and output looks a bit nicer)

```diff
diff --git i/cli/Cargo.toml w/cli/Cargo.toml
index 544e573..5719bec 100644
--- i/cli/Cargo.toml
+++ w/cli/Cargo.toml
@@ -7,11 +7,10 @@ license = "MIT OR Apache-2.0"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

 [dependencies]
-clap = { version = "3.0.6", features = [
+clap = { version = "3.0.6", default-features = false, features = [
+    "suggestions",
     "derive",
     "env",
-    "unicode",
-    "wrap_help",
 ] }
 clap_complete = "3.0.2"
 foundry-utils = { path = "../utils" }
```

**Review by gakonst — APPROVED** [review]
looks gerat

File: README.md [hunk 1]
```diff
@@ -90,6 +90,19 @@ cast call 0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48 "totalSupply()(uint256)" --
...
+
+You can generate autocompletion shell scripts for bash, elvish, fish, powershell, and zsh.
+
+Example (zsh / [oh-my-zsh](https://ohmyz.sh/))
+
+```
+mkdir -p ~/.oh-my-zsh/completions
+forge completions zsh > ~/.oh-my-zsh/completions/_forge
+cast completions zsh > ~/.oh-my-zsh/completions/_cast
+source ~/.vimrc
```
Comment (gakonst) [inline README.md:103]:
```suggestion
source ~/.zshrc
```

**Review by gakonst — APPROVED** [review]

**Comment by gakonst** [comment]
Life changer:

```bash
forge completions zsh > ~/.oh-my-zsh/completions/_forge
cast completions zsh > ~/.oh-my-zsh/completions/_cast
source ~/.zshrc
```

**Pull request merged**